dvc/scm/__init__.py

```diff
@@ -21,7 +23,12 @@ def SCM(root_dir):  # pylint: disable=invalid-name
     Returns:
         dvc.scm.base.Base: SCM instance.
     """
-    if Git.is_repo(root_dir) or Git.is_submodule(root_dir):
-        return Git(root_dir)
 
-    return NoSCM(root_dir)
+    try:
+        scm = Git(
+            root_dir, search_parent_directories=search_parent_directories
+        )
+    except SCMError:
+        scm = NoSCM(root_dir)
```

Comment:
If we explicitly have `no-scm = True` in our config, we probably shouldn't try to search for git, as that might defeat the purpose :)

Reply:
So maybe we might use NoSCM directly in `Repo.__init__` if `no-scm` instead of going through `SCM` wrapper?

Reply:
Seems like we can `search_parent_directories=True` by default (in Git class itself) and replace it with `no_scm` flag here. Might be missing something, but right now I see that it is only used for that purpose but in an obscure way, so I suppose these are the leftovers from the previous implementation, that you didn't reconsider.

Reply:
I see one problem here: what about `Repo.init` method? If we will rely on `no_scm` there, then there is no point of having `--subdir` flag.
I believe it should stay as is because in `Repo.init` we have 3 situations to handle (and thats why we need 2 flags):
1. no-scm=True, subdir=False -> succesfull init
2. no-scm=False, subdir=False + NoSCM -> raise error
3. no-scm=False, subdir=True -> search parent dirs

Reply:
@pared Agreed on that. So `search_parent_directories=False` should be used in `repo.init()` if `no_scm=False` and `subdir=False`, right? In all other cases it should be `search_parent_directories=True`, right?

Reply:
@efiop that is correct

Reply:
@efiop, now logic related to choosing SCM is in `Repo.__init__`. Now I am not sure whether we need `SCM` method at all. I believe we could refactor `init` command to not use it. Do you think its still the scope of this task?